CONTRIBUTING.md

```diff
@@ -0,0 +1,50 @@
+# Contributing Guide
+
+This guide should help you get started with contributing to the project. It will cover things such as getting your development environment setup and how to run the tests.
+
+## Development Environment
+
+This project uses [`hatch`](https://hatch.pypa.io/latest/) as its project management tool.
+
+To enter the development environment (it will be created automatically) run:
+
+```bash
+$ hatch shell
+```
+
+Once in this environment you can run `condax` commands to see how your current code behaves. If you don't run `hatch shell`, you should prefix all commands mentioned below with `hatch run`.
+
+## Running Tests
+
+To run the tests you can use `pytest`.
+
+```bash
+$ pytest
+```
+
+## Code Style
+
+This project uses `black` for code formatting and `isort` to order imports. To format your code run:
+
+```bash
+$ black .
+$ isort .
+```
+
+## Type Checking
+
+This project uses `mypy` for type checking. To run the type checker run:
+
+```bash
+$ mypy .
+```
+
```

Comment:
I think it's easiest to leave style, formatting, and type checking to the pre-commit hook. It's way fewer steps for the contributor and happens automatically. You could replace these sections with a single section explaining how to install pre-commit.